### ISSUE TITLE
Ping-pong pairing

### ISSUE THREAD

**Issue by sofer** [FAC6/FAC7#13]
What do people think about introducing a more rigorous approach to programming in the classroom, in particular mandating [ping-pong pairing](http://c2.com/cgi/wiki?PairProgrammingPingPongPattern)?

**Comment by sofer** [comment]
See also http://articles.coreyhaines.com/posts/thoughts-on-pair-programming/

**Comment by sofer** [comment]
Also, how do people feel about insisting that pairs start by writing out a potential solution on paper first?

**Comment by jackcarlisle** [comment]
I think both the ping-pong introduction and the paper solutions ideas are great! I've found personally that by writing a solution down on paper first allows you to organise your thoughts in a clearer way. I would usually write the solution in plain English first and then translate it into code.
